### PR TITLE
Add multi-field search support to value filter.

### PR DESCRIPTION
**Pull request by ndm2**
Basically a copy of #135, but complete with tests.

However, I've just noticed that having a mode for the values doesn't seem to make any sense, given that this filter applies an EQ comparison, applying AND to multiple values would never match, as a field can only hold one of the values in such a comparison.

Am I missing something, or did I just made a mistake when I added this initially, and nobody noticed it?

### CODE REVIEW

**Comment by ADmad** [comment]
> However, I've just noticed that having a mode for the values doesn't seem to make any sense, given that this filter applies an EQ comparison, applying AND to multiple values would never match, as a field can only hold one of the values in such a comparison.

You are right, using `AND` for multiple values make sense, it would never match. Seems we did over look that.

**Comment by dereuromark** [comment]
Yes, it only makes sense for the field part.

**Comment by ndm2** [comment]
Given that it would be kind of a bug fix, would you be OK with a "hard break" in terms of removing that feature/option, ie only use `IN()` for multiple values?

**Comment by dereuromark** [comment]
Lets skip it together with https://github.com/FriendsOfCake/search/pull/156 as major bump. All good then.

**Comment by ADmad** [comment]
I don't think we are breaking anything as `'mode' => 'AND' `  would have never "worked" in the first place.

**Comment by ndm2** [comment]
That would be my stance on it too, just wanted to ask as people have different opinions when it comes to removing not-correctly/non working, but possibly in use behavior. I'll remove it.

File: tests/TestCase/Model/Filter/ValueTest.php [hunk 1]
```diff
@@ -215,13 +317,13 @@ public function testProcessNoDefaultFallbackForDisallowedMultiValue()
...
      */
-    public function testProcessCaseInsensitiveMode()
+    public function testProcessCaseInsensitiveValueMode()
     {
         $articles = TableRegistry::get('Articles');
         $manager = new Manager($articles);
         $filter = new Value('title', $manager, [
             'multiValue' => true,
-            'mode' => 'Or'
+            'valueMode' => 'Or'
```
Comment (dereuromark) [inline tests/TestCase/Model/Filter/ValueTest.php]: above you used "mode".

**Comment by dereuromark** [comment]
@ADmad Do you want to tag a new minor before we merge and release the new major change?

**Pull request merged**